draft-ietf-httpbis-digest-headers.md

```diff
@@ -77,442 +76,385 @@ normative:
-at rest.
-They can be used across multiple hops in order to provide end-to-end
-integrity guarantees, which can aid fault diagnosis when resources are
-transferred across hops and system boundaries.
-Finally, they can be used to
-validate integrity when reconstructing a resource fetched using different HTTP
-connections.
-
-This document obsoletes [RFC3230].
+
+This document obsoletes RFC 3230 and therefore the Digest and Want-Digest HTTP
+fields; see {{obsolete-3230}}.
+
 
 ## Document Structure
 
 This document is structured as follows:
 
-- {{digest}} describes concepts related to representation digests. It also
-  defines the Digest request and response header and trailer field,
+- {{representation-digest}} defines the Representation-Digest request and response header and trailer field,
 - {{content-digest}} defines the Content-Digest request and response header and trailer field,
-- {{want-fields}} defines the Want-Digest and Want-Content-Digest request and response header and
+- {{want-fields}} defines the Want-Representation-Digest and Want-Content-Digest request and response header and
   trailer field,
-- {{algorithms}} describes algorithms and their relation to Digest,
+- {{algorithms}} describes algorithms and their relation to the fields defined in this document,
 - {{state-changing-requests}} details computing representation digests,
 - {{examples-unsolicited}} and {{examples-solicited}} provide examples of using
-  Digest and Want-Digest.
+  Representation-Digest and Want-Representation-Digest.
 
 ## Concept Overview
 
-This document defines the `Digest` request and response header and trailer
-field; see {{digest}}. At a high level, the value contains a checksum, computed
-over `selected representation data` ({{Section 3.2 of
-SEMANTICS}}), that the recipient can use to
-validate integrity. Basing `Digest` on the selected representation makes it
-straightforward to apply it to use-cases where the transferred data requires
-some sort of manipulation to be considered a representation or conveys a
-partial representation of a resource, such as Range Requests (see {{Section
-14.2 of SEMANTICS}}).
-
-To support use-cases where a simple checksum of the content bytes is required,
-this document introduces the `Content-Digest` request and response header and trailer field; see
+The HTTP fields defined in this document can be used for HTTP integrity. Senders
+choose a hashing algorithm and calculate a digest from an input related to the
+HTTP message, the algorithm identifier and digest are transmitted in an HTTP
+field. Receivers can validate the digest for integrity purposes. Hashing
+algorithms are registered in the "Hash Algorithms for HTTP Digest Fields" (see
+{{algorithms}}).
+
+Selecting the data on which digests are calculated depends on the use case of
+HTTP messages. This document provides different headers for HTTP representation
+data and HTTP content.
```

Comment:
These concepts really need to be defined.  Probably right here.

Reply:
@martinthomson these concepts are defined in SEMANTICS and referenced in Notational Convention. There's an Annex explaining this stuff too. Am I missing something?

To ease this kind of work, I filed an issue on kramdown-rfc2629 that could help simplifying referencing concepts when using markdown. https://github.com/cabo/kramdown-rfc2629/issues/155

Reply:
I was looking for a citation right here.  It seemed like the right place for it.  Given how critical they are, it seems prudent to cite the relevant sections of {{SEMANTICS}} right here.

Reply:
The HTML makes this easier to reason about. In https://httpwg.org/http-extensions/draft-ietf-httpbis-digest-headers.html#name-introduction. We cite the relevant sections in the next paragraphs. Part of the editorial work on reordering sections will refactoring things a little and sort out citing at the earliest and kost convenient time